### PR TITLE
Fixes completion of files with spaces and adds file type identifier

### PR DESCRIPTION
**Pull request by JanValiska**
Related issue: #55

### CODE REVIEW

File: lua/source/path.lua [hunk 1]
```diff
@@ -83,8 +93,9 @@ M.triggerFunction = function(_, _, _, manager)
   local stdout = vim.loop.new_pipe(false)
   local stderr = vim.loop.new_pipe(false)
   local handle, pid
-  handle, pid = vim.loop.spawn('ls', {
-    args = {path, '-A'},
+  print(path)
```
Comment (haorenW1025) [inline lua/source/path.lua:96]: Hey this is so nice! Can you get rid of the `print` though? I accidentally left that in and it should be remove.

Comment (JanValiska) [reply lua/source/path.lua]: Hi, print removed.

**Comment by haorenW1025** [comment]
I've found a weird issue though, sometimes the return type will be `m` and it will cause error. I can reproduce this using path completion and type `/usr/bin/`.

**Comment by haorenW1025** [comment]
Maybe use a metatable to prevent from getting nil value? Something like this
```lua
local fileTypesMap = setmetatable({
    ['f'] = "(file)",
    ['d'] = "(dir)",
    ['c'] = "(char)",
    ['l'] = "(link)",
    ['b'] = "(block)",
    ['p'] = "(pipe)",
    ['s'] = "(socket)"
}, {__index = function() 
    return ''
  end
})
```

**Comment by JanValiska** [comment]
Using suggested metatable solved the issue but even that i will get two empty words in complete_items list. I will look at it later.

**Comment by haorenW1025** [comment]
I did notice there is some stuff in completion list in that case, and when I'm using `find /usr/bin/ -mindepth 1 -maxdepth 1 -printf '%y'` in the terminal all things are `l` or `f`..., pretty weird.
Or an alternative is simply skipping the weird stuff out, they should be meaningless anyway.
```lua
  for _, val in ipairs(M.items) do
    local score = score_func(prefix, val.name)
    if score < #prefix/3 or #prefix == 0 then
      if fileTypesMap[val.t] then goto continue end
      table.insert(complete_items, {
        word = val.name,
        kind = 'Path ' .. fileTypesMap[val.t],
        score = score,
        icase = 1,
        dup = 1,
        empty = 1,
      })
    end
    ::continue::
  end
```

**Comment by JanValiska** [comment]
I rewrote directory listing using uv fs_scandir function. It is working better now. But there is still little problem with completion of items with spaces. 

Try create this directory structure `mkdir -p "dir_one/dir two/dir_three"` and `touch "dir_one/dir two/file"`, then start completing `dir_one/` and try to reach `file`. It still doesn't work.

**Comment by haorenW1025** [comment]
Hmm I make some test and I think the cause is this line
```vim
  local textMatch = vim.fn.match(line_to_cursor, '\\f*$')
```
I use the vim built-in matching function to match the file name structure, but it can't recognize file name with spaces. Maybe we have to rewrite this.

**Comment by haorenW1025** [comment]
I think I'll merged this first since using `fs_scandir` is definitely a better choice. After that I'll rewrite bunch of stuff(since the code was not clear at all) to get this working with white spaces in path name. Will notify you when I finish in the issues. Thanks a lot for the contribute still:)

**Pull request merged**